starknet_py/contract_test.py

```diff
@@ -139,3 +141,46 @@ def test_deploy_result_post_init(gateway_client):
             hash=0,
             _client=gateway_client,
         )
+
+
+def test_contract_raises_on_no_provider_and_client():
+    with pytest.raises(ValueError, match="One of provider or client must be provided."):
+        Contract(address=1234, abi=[])
+
+
+def test_contract_raises_on_both_provider_and_client(
+    gateway_account_client, gateway_client
+):
+    with pytest.raises(
+        ValueError, match="Arguments provider and client are mutually exclusive."
+    ):
+        Contract(
+            address=1234, abi=[], provider=gateway_account_client, client=gateway_client
+        )
+
+
+def test_contract_create_with_base_account(gateway_account):
+    contract = Contract(address=0x1, abi=[], provider=gateway_account)
+    assert isinstance(contract.account, BaseAccount)
+    assert contract.account == gateway_account
+    assert contract.client == gateway_account.client
+
+
+def test_contract_create_with_account_client(gateway_account_client):
+    contract = Contract(address=0x1, abi=[], provider=gateway_account_client)
+    assert isinstance(contract.account, AccountProxy)
+    assert contract.client == gateway_account_client.client
+
+    contract = Contract(address=0x1, abi=[], client=gateway_account_client)
+    assert isinstance(contract.account, AccountProxy)
+    assert contract.client == gateway_account_client.client
```

Comment:
That could be parametrized, but it will require using the `gateway_account_client` fixture in the `@pytest.mark.parametrize` which is not simple.

Reply:
It is ok as it is

Reply:
I'm not a fan of tests like this either. These could be their own separate tests, but `client` (and its tests) is going to be removed at some point, so I think it's fine to have it like this as a temporary solution.